### PR TITLE
Bundle library dependencies in macOS release workflow

### LINKED ISSUE #49570: 0.F (tiles) does not launch on macOS
# Describe the bug

Double-clicking on the Mac app does not launch the game. The app icon bounces a few times, and then nothing. If I "Show Package Contents" and then double-click on the `cataclysm-tiles` binary, Terminal opens and tells me:

```
dyld: Library not loaded: /usr/local/opt/sdl2_ttf/lib/libSDL2_ttf-2.0.0.dylib
  Referenced from: /Applications/Cataclysm.app/Contents/Resources/cataclysm-tiles
  Reason: image not found
```

I can see that this dylib is in the app, but the binary isn't looking for it there.

Installing `sdl2_ttf` via brew allows the app to launch.

~~(though I'm getting error messages related to `json.cpp`, but I'll report on that separately...~~

(Removing my comment about `json.cpp`, since this was due to a mistake I made installing 0.F. Basically, if I install 0.F without first uninstalling 0.E, it leaves files in ~/Library/Application Support/Cataclysm that cause error messages. Uninstalling 0.E, deleting that folder, and then installing 0.F fixes the `json.cpp` issues I was seeing.)

# Steps To Reproduce

1. Double click on Cataclysm.app

# Expected behavior

The game loads.

# Versions and configuration

- OS: MacOs
    - OS Version: Name: macOS; Version: 11.4; Build: 20F71; 
- Game Version: bc28921 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No name () []
]

### LINKED ISSUE #50198: mac os ox build for 0.F is missing libSDL2mixer-2.0.0 dylib file and wont start
### Describe the bug

mac os ox build for 0.F is missing libSDL2 mixer-2.0.0 dylib file and won't run! bescue there's only half of the libsdl2 mixer files needed for normal function in the games core resource folder! i can't run the game without the other half of the sdl2 mixer files that not in the game files! it looks like someone forgot one needed file for the mac os x 0.F build of CDDA to work!

### Steps To Reproduce

trying to start up the game
then it crashes

### Expected behavior

the main menu should pop up

### Screenshots

_No response_

### Versions and configuration

mac os x 10.13.6

### Additional context

Dyld Error Message:
  Library not loaded: /usr/local/opt/sdl2_mixer/lib/libSDL2_mixer-2.0.0.dylib
  Referenced from: /Applications/Cataclysm 7.app/Contents/Resources/./cataclysm-tiles
  Reason: image not found

### PR DESCRIPTION
**Pull request by BrettDong**
#### Summary
Build "Bundle library dependencies in macOS release workflow"

#### Purpose of change
Fix #49570.
Fix #50198.

In the new GitHub Action release workflow, the executable is linked against SDL2 and audio codec libraries installed at `/usr/local/opt` from Homebrew package manager, and references libraries at `/usr/local/opt/` on build machine rather than the ones bundled in the release distribution. This leads to "missing dylib" error when trying to launch the game on a different system, as described in #49570 and #50198.

#### Describe the solution
1. Invoke `otool -L` command to list dynamic library dependencies of the executable
2. `grep` the result from step (1) with either the path from `$(sdl2-config --libs)`, `/usr/local/opt` (the old Homebrew location) or `/opt/homebrew` (the new Homebrew location) to extract the paths of libraries to be copied
3. Copy those libraries to release distribution folder
4. Invoke `install_name_tool` command to rewrite dynamic library references from absolute path like `/opt/homebrew/opt/sdl2_ttf/lib/libSDL2_ttf-2.0.0.dylib` to relative path like `@executable_path/libSDL2_ttf-2.0.0.dylib`.

#### Describe alternatives you've considered
Not really.

#### Testing
The executable now references libraries shipped in release bundle:
```
otool -L Cataclysm.app/Contents/Resources/cataclysm-tiles 
Cataclysm.app/Contents/Resources/cataclysm-tiles:
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1775.118.101)
	@executable_path/libintl.8.dylib (compatibility version 11.0.0, current version 11.0.0)
	/System/Library/Frameworks/Cocoa.framework/Versions/A/Cocoa (compatibility version 1.0.0, current version 23.0.0)
	@executable_path/libSDL2-2.0.0.dylib (compatibility version 15.0.0, current version 15.0.0)
	@executable_path/libSDL2_ttf-2.0.0.dylib (compatibility version 15.0.0, current version 15.1.0)
	@executable_path/libSDL2_image-2.0.0.dylib (compatibility version 3.0.0, current version 3.3.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 905.6.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.100.5)
```

### CODE REVIEW

**Comment by actual-nh** [comment]
So that people can use `install_name_tool` to switch SDL2 library versions, etc, perhaps `ld` should be run with the `-headerpad_max_install_names` option to build binaries?

**Comment by BrettDong** [comment]
Technically yes people can change SDL library references with `install_name_tool` but I don't think this should be encouraged as things could probably break.

File: Makefile [hunk 1]
```diff
@@ -1119,10 +1119,12 @@ ifdef LANGUAGES
 	mkdir -p $(APPRESOURCESDIR)/lang/mo/
 	cp -pR lang/mo/* $(APPRESOURCESDIR)/lang/mo/
 endif
+ifeq ($(TILES), 0)
 ifeq ($(LOCALIZE), 1)
 	LIBINTL=$$($(CROSS)otool -L $(APPTARGET) | grep libintl | sed -n 's/\(.*\.dylib\).*/\1/p') && if [ -f $$LIBINTL ]; then cp $$LIBINTL $(APPRESOURCESDIR)/; fi; \
```
Comment (actual-nh) [inline Makefile:1124]: This actually needs to be switched to reference the libintl in the executable directory - see my comment in #49572 - at least to enable command-line execution when desired.

**Comment by actual-nh** [comment]
> Technically yes people can change SDL library references with `install_name_tool` but I don't think this should be encouraged as things could probably break.

It certainly shouldn't replace an adequate patch for SDL library incompatibilities, but it can serve as a workaround in the meantime. (Future-proofing.) It would also be nice for switching them for development purposes, I suspect (I don't use SDL, so I don't know for sure).

**Pull request merged**